ui/tests/test_importer_views.py

```diff
@@ -159,7 +161,44 @@ def test_invalid_form(self):
             self.import_url_slug,
             {}, follow=True
         )
-        self.assertTrue(resp.status_code == HTTP_OK)
-        body = resp.content.decode("utf-8")
-        log.debug(body)
-        self.assertTrue("This field is required." in body)
+        self.assertContains(resp, "This field is required.")
+
+    def test_wiped_index(self):
+        """
+        If the Elasticsearch index is erased, it caused a KeyError
+        when accessing the repository page until Django is restarted.
+        """
+        # Site is returning a 200, and there's known data in the output.
+        resp = self.client.get(self.repository_url_slug, follow=True)
+        self.assertContains(resp, 'Febtober')
+
+        # Delete the Elasticsearch index.
+        call_command('clear_index', interactive=False, verbosity=0)
+
+        # Save to restore after this test.
+        old_func = FacetedSearchView.extra_context
+
+        def extra_context(self):
+            """
+            Simulate results of deleting the index while the system
+            is running. Specifically, the "fields" key will be missing
+            from the "facets" dict.
+            """
+            results = old_func(self)
```

Comment:
Note that this function, as-is, works. If we leave line 195 commented out, everything is fine and all tests pass. We just don't have coverage of the new `if` statement in `ui/views.py`.